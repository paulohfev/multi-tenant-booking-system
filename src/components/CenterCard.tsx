import Image from 'next/image'
import Link from 'next/link'
import { Center } from '@/types/Center'

export type CenterCardProps = {
  center: Center
}

const CenterCard: React.FC<CenterCardProps> = ({ center }) => {
  return (
    <Link key={center.id} href={`/centers/${center.id}`}>
      <div className='flex flex-col bg-white rounded-lg shadow-md p-6 hover:shadow-lg transition-shadow duration-200 h-[320px]'>
        <div className='relative w-20 h-20 mb-4'>
          <Image src={center.logo} alt={`${center.name} logo`} fill className='object-contain' />
        </div>
        <h3 className='text-xl font-semibold text-gray-900 mb-2'>{center.name}</h3>
        <p className='text-gray-600'>{center.description}</p>
        <div className='mt-auto'>
          <span className='inline-flex items-center px-3 py-1 rounded-full text-sm font-medium bg-blue-100 text-blue-800'>
            {center.services} services available
          </span>
        </div>
      </div>
    </Link>
  )
}

export default CenterCard
