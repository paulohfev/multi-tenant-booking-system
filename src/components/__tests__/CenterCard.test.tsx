import { render } from '@testing-library/react'
import CenterCard from '../CenterCard'

const mockCenter = {
  id: '1',
  name: 'Test Center',
  description: 'Test Description',
  logo: '/logos/test-center.svg',
  services: 5,
}

describe('CenterCard', () => {
  it('should match snapshot', () => {
    const { container } = render(<CenterCard center={mockCenter} />)
    expect(container).toMatchSnapshot()
  })
})
