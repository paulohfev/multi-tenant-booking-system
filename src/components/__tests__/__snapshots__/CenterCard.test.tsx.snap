// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterCard should match snapshot 1`] = `
<div>
  <a
    href="/centers/1"
  >
    <div
      class="flex flex-col bg-white rounded-lg shadow-md p-6 hover:shadow-lg transition-shadow duration-200 h-[320px]"
    >
      <div
        class="relative w-20 h-20 mb-4"
      >
        <img
          alt="Test Center logo"
          class="object-contain"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          src="/logos/test-center.svg"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <h3
        class="text-xl font-semibold text-gray-900 mb-2"
      >
        Test Center
      </h3>
      <p
        class="text-gray-600"
      >
        Test Description
      </p>
      <div
        class="mt-auto"
      >
        <span
          class="inline-flex items-center px-3 py-1 rounded-full text-sm font-medium bg-blue-100 text-blue-800"
        >
          5
           services available
        </span>
      </div>
    </div>
  </a>
</div>
`;
